src/core/core.scale.js

```diff
@@ -1607,13 +1609,23 @@ export default class Scale extends Element {
 		const items = me._labelItems || (me._labelItems = me._computeLabelItems(chartArea));
 		let i, ilen;
 
+		ctx.save();
+		if (position === 'left' || position === 'right') {
+			ctx.rect(me.left, 0, me.width, me.chart.height);
```

Comment:
0 does not work when there are multiple scales on the same position. 
There are `clipArea` and `unclipArea` helpers for this.

Reply:
Updated with using the label area. Still needed some adjustments because labels end up outside that area (rotation or alignment), so let me know what you think of that approach.